src/radical/entk/appman/wfprocessor.py

```diff
@@ -100,145 +110,173 @@ def _initialize_workflow(self):
+
+                                local_transition(obj=exec_stage,
+                                                    obj_type='Stage',
+                                                    new_state=states.SCHEDULING,
+                                                    profiler=self._prof,
+                                                    logger=self._logger,
+                                                    reporter=self._report)
+
+                            exec_tasks = exec_stage.tasks
+
+                        for exec_task in exec_tasks:
+
+                            if (exec_task.state == states.INITIAL) or \
+                                ((exec_task.state == states.FAILED) and
+                                    (self._resubmit_failed)):
+
+                                # Set state of Tasks in current Stage
+                                # to SCHEDULING
+                                local_transition(obj=exec_task,
+                                                    obj_type='Task',
+                                                    new_state=states.SCHEDULING,
+                                                    profiler=self._prof,
+                                                    logger=self._logger,
+                                                    reporter=self._report)
+
+                                # Store the tasks from different pipelines
+                                # into our workload list. All tasks will
+                                # be submitted in bulk and their states
+                                # will be updated accordingly
+                                workload.append(exec_task)
+
+                                # We store the stages since the stages the
+                                # above tasks belong to also need to be
+                                # updated. If its a task that failed, the
+                                # stage is already in the correct state
+                                if exec_task.state == states.FAILED:
+                                    continue
+                                if exec_stage not in scheduled_stages:
+                                    scheduled_stages.append(exec_stage)
+
+                # If there are tasks to be scheduled
                 if workload:
 
-                    # Put the task on one of the pending_queues
-                    workload_as_dict = list()
+                    # Tasks of the workload need to be converted into a dict
+                    # as pika can send and receive only json/dict data
+                    workload_as_json = list()
                     for task in workload:
-                        workload_as_dict.append(task.to_dict())
-                    workload_as_dict = json.dumps(workload_as_dict)
+                        workload_as_json.append(task.to_dict())
+                    workload_as_json = json.dumps(workload_as_json)
 
```

Comment:
This could be written as 
```
  wl_json = [task.to_dict() for task in workload]
```
or even:
```
  wl_json = json.dumps([task.to_dict() for task in workload])
```
if you like that kind of thing...

Reply:
Yup, makes sense :+1: